tests/operator/operator.go

```diff
@@ -949,8 +950,10 @@ spec:
 			Expect(kv.GetGeneration()).To(Equal(generation))
 
 			By("Deleting patch from KubeVirt object")
-			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
-			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(context.Background(), kv, metav1.UpdateOptions{})
+
+			patchPayload, err = patch.New(patch.WithReplace("/spec/customizeComponents", v1.CustomizeComponents{})).GeneratePayload()
```

Comment:
looks good to me. Just wander if it can be done by `{"op": "delete"}`

Reply:
switched to `patch.WithRemove`, should be fine I guess it generates the same behavior :+1:

Reply:
Tested it manually. before the test, `customizeComponents` is an empty object (`{}`), but after the patch it's nil. I think it's ok but not sure.

Reply:
i traced the code to this point

```
func getHash(customizations v1.CustomizeComponents) (string, error) {
	// #nosec CWE: 326 - Use of weak cryptographic primitive (http://cwe.mitre.org/data/definitions/326.html)
	// reason: sha1 is not used for encryption but for creating a hash value
	hasher := sha1.New()

	sort.SliceStable(customizations.Patches, func(i, j int) bool {
		return len(customizations.Patches[i].Patch) < len(customizations.Patches[j].Patch)
	})

	values, err := json.Marshal(customizations)
	if err != nil {
		return "", err
	}
	hasher.Write(values)

	return hex.EncodeToString(hasher.Sum(nil)), nil
}
```

here if the field is nil, we should hit nil pointer exception.. while on the other hand if it's initialized to empty struct the length of the patches field will be initialized to 0

Reply:
That ok. The `.spec.customizeComponents` is not a pointer

Reply:
done